app/src/main/java/com/pvp/app/ui/screen/steps/StepViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.pvp.app.ui.screen.steps
+
+import androidx.health.connect.client.records.StepsRecord
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.health.connect.client.HealthConnectClient
+import androidx.health.connect.client.PermissionController
+import androidx.health.connect.client.permission.HealthPermission
+import androidx.health.connect.client.records.ExerciseSessionRecord
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.pvp.app.api.HealthConnectService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import javax.inject.Inject
+
+@HiltViewModel
+class StepViewModel @Inject constructor(
+    private val healthConnectClient: HealthConnectClient,
+    private val healthConnectService: HealthConnectService
+) : ViewModel() {
+
+    val PERMISSIONS = setOf(
+        HealthPermission.getReadPermission(StepsRecord::class),
+        HealthPermission.getReadPermission(ExerciseSessionRecord::class)
+    )
+
+    private val _stepsCount = MutableStateFlow(0)
+    val stepsCount = _stepsCount.asStateFlow()
+
+    val permissionsLauncher = PermissionController.createRequestPermissionResultContract()
+
+    suspend fun permissionsGranted(): Boolean {
+        val granted = healthConnectClient.permissionController.getGrantedPermissions()
+        return granted.containsAll(PERMISSIONS)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
+    fun updateTodaysSteps() {
+        viewModelScope.launch {
+            val endTime = Instant.now()
+            val startTime = LocalDate.now().atStartOfDay(ZoneId.systemDefault()).toInstant()
```

Comment:
Names `end` and `start` would be quite understandable too. For as long as there is no other context for `end/start` names. If there were `endDate`, `startDate` together with these times, then `time` and `date` specifiers should stay.